lib/matplotlib/tests/test_ticker.py

```diff
@@ -613,3 +613,175 @@ def test_latex(self, is_latex, usetex, expected):
+
+    def test_attributes(self, empty_fmt):
+        # Using == is the right way to compare bound methods:
+        # http://stackoverflow.com/q/41900639/2988730
+        assert empty_fmt.transform == self.transform1
+        assert isinstance(empty_fmt.formatter, mticker.ScalarFormatter)
+
+    def test_create_dummy_axis(self, empty_fmt):
+        assert empty_fmt.axis is None
+        assert empty_fmt.formatter.axis is None
+
+        empty_fmt.create_dummy_axis()
+
+        assert isinstance(empty_fmt.axis, mticker._DummyAxis)
+        assert isinstance(empty_fmt.formatter.axis, mticker._DummyAxis)
+        assert empty_fmt.axis is not empty_fmt.formatter.axis
+
+    def test_set_axis(self, fmt):
+        prev_axis = fmt.axis
+        prev_inner_axis = fmt.formatter.axis
+
+        fmt.set_axis(mticker._DummyAxis())
+
+        assert fmt.axis is not None
+        assert fmt.axis is not prev_axis
+        assert fmt.formatter.axis is prev_inner_axis
+
+        fmt.set_axis(None)
+        assert fmt.axis is None
+        assert fmt.formatter.axis is None
+
+    def test_set_view_interval(self, fmt):
+        bounds = [100, 200]
+        xbounds = [self.transform1(x) for x in bounds]
+
+        fmt.set_view_interval(*bounds)
+
+        assert np.array_equal(fmt.axis.get_view_interval(), bounds)
+        assert np.array_equal(fmt.formatter.axis.get_view_interval(),
+                              xbounds)
+
+    def test_set_data_interval(self, fmt):
+        bounds = [50, 60]
+        xbounds = [self.transform1(x) for x in bounds]
+
+        fmt.set_data_interval(*bounds)
+
+        assert np.array_equal(fmt.axis.get_data_interval(), bounds)
+        assert np.array_equal(fmt.formatter.axis.get_data_interval(), xbounds)
+
+    def test_set_bounds(self, fmt):
+        bounds = [-7, 7]
+        xbounds = [self.transform1(x) for x in bounds]
+
+        fmt.set_bounds(*bounds)
+
+        assert np.array_equal(fmt.axis.get_view_interval(), bounds)
+        assert np.array_equal(fmt.axis.get_data_interval(), bounds)
+        assert np.array_equal(fmt.formatter.axis.get_view_interval(), xbounds)
+        assert np.array_equal(fmt.formatter.axis.get_data_interval(), xbounds)
+
+    def test_format_data(self, fmt):
+        with matplotlib.rc_context({'text.usetex': False}):
+            assert fmt.format_data(100.0) == '\N{MINUS SIGN}1e2'
+
+    def test_format_data_short(self, fmt):
+        assert fmt.format_data_short(-200.0) == '{:<12g}'.format(200)
+
+    def test_get_offset(self, fmt):
+        assert fmt.get_offset() == fmt.formatter.get_offset()
+
+    def test_set_locs(self, fmt):
+        locs = [1.0, 2.0, 3.0]
+        xlocs = [self.transform1(x) for x in locs]
+
+        fmt.set_locs(locs)
+
+        # Currently, `fmt.locs is locs` works, but should not be
+        # tested for since it is not contractually guaranteed.
+        assert fmt.locs == locs
+        assert fmt.formatter.locs == xlocs
+
+        fmt.set_locs(None)
```

Comment:
this is  a teeny test, but it's a different one `def test_clear_locs` or something